src/clj/fluree/db/json_ld/commit_data.cljc

```diff
@@ -524,6 +522,14 @@
   [t commit-sid message]
   [(flake/create commit-sid const/$_commit:message message const/$xsd:string t true nil)])
 
+;; TODO - txn should really be an IRI, not a string
```

Comment:
We persist it to disk based on its unique hash-based id (content-addressed). So the @id should be the unique identifier, and if we follow the pattern we already have established for commits and data files, we would also include "f:address" as the physical storage location(s).

On disk would be JSON-LD looking something like:
```
{"@id": "fluree:txn:sha256:...." ;; eg IPFS IRI address, local file, s3
 "@type": ["f:Transaction"]
 "f:address" "...some physical storage location addres ..."
 "f:tx-raw" ".... stringified raw data ..."}  ;; this could be JSON, JWS, or even some byte data for different crypto
```
It could also include other metadata with it... e.g. for a different type of crypto it might include a "f:verificationProof", or whatever... it just just JSON-LD and infinitely flexible. Technically it doesn't have to include everything above either, as some could be inferred (if we were super storage-concerned).

So for the purposed of the commit file, it would likely only have the {"@id": "...", "f:address": "..."} embedded to minimize space.... any other metadata (if any besides the raw txn) would have to go to disk to retrieve.

This is a bigger change of course, which is why I left this TODO.

Reply:
This makes sense. 👍🏾